### PR TITLE
Adds write-to-file flag to store data locally

### PR DESCRIPTION
**Pull request by JammyL**
Flag --write-to-file allows for a local path to be given to which data is stored.
Only stores data locally if no organization is specified. - This can be added for when organisation is specified if needed.

Signed-off-by: Jamie Leppard <JammyL@users.noreply.github.com>

### CODE REVIEW

File: cmd/agent.go [hunk 1]
```diff
@@ -64,5 +64,12 @@ func init() {
 		false,
 		"Runs agent a single time if true, or continously if false",
 	)
+	agentCmd.PersistentFlags().StringVarP(
+		&agent.DataFilePath,
+		"write-to-file",
```
Comment (charlieegan3) [inline cmd/agent.go:69]: Do you think `--output-path` might be more descriptive?

File: cmd/agent.go [hunk 2]
```diff
@@ -64,5 +64,12 @@ func init() {
...
+		"",
+		"",
+		"Data file path, if used will write data to a local file instead of posting",
```
Comment (charlieegan3) [inline cmd/agent.go:72]: `...instead of posting` -. `...instead of uploading to the preflight server`

File: pkg/agent/run.go [hunk 3]
```diff
@@ -176,27 +178,35 @@ func gatherAndPostData(ctx context.Context) {
 	log.Println("Running Agent...")
 	log.Println("Posting data to ", baseURL)
 	if config.OrganizationID == "" {
```
Comment (charlieegan3) [inline pkg/agent/run.go:180]: I think we might need to handle the case when the org is set too?

File: pkg/agent/run.go [hunk 4]
```diff
@@ -176,27 +178,35 @@ func gatherAndPostData(ctx context.Context) {
...
-		data, err := json.Marshal(readings)
+		data, err := json.MarshalIndent(readings, " ", " ")
```
Comment (charlieegan3) [inline pkg/agent/run.go:181]: Ideally, we only indent the json when we're writing to disk. Data sent over the wire should be as small as possible.

File: cmd/agent.go (hunk 1 above)
Comment (JammyL) [reply cmd/agent.go:69]: Yes, that's a better name

File: pkg/agent/run.go (hunk 3 above)
Comment (JammyL) [reply pkg/agent/run.go:180]: Yes, I think this should be a part of the larger changes made with the load-from-file addition

**Pull request merged**